src/magnet_link_parse_error.rs

```diff
@@ -0,0 +1,21 @@
+use crate::common::*;
+
+#[derive(Debug, Snafu)]
+#[snafu(visibility(pub(crate)))]
+pub(crate) enum MagnetLinkParseError {
+  #[snafu(display(
+    "Peer address `{}` failed to parse into a (host:port): {}",
+    text,
+    source
+  ))]
+  Peer {
+    text: String,
+    source: HostPortParseError,
+  },
+  #[snafu(display("Magnet links must begin with `magnet:`"))]
```

Comment:
```
Invalid scheme `{}`, magnet links must use the `magnet:` scheme
```

Reply:
Something about the comma looks strange. Sure you want this exact message?

Reply:
I'm not wedded to the exact message, as long as it contains the invalid scheme. (Since I'm not sure if all users will know what a URL scheme is.)